src/Controller/UI/ReportResearchGroupDatasetStatusController.php

```diff
@@ -114,28 +113,31 @@ public function datasetMonitoringReportAction(Request $request, EntityHandler $e
 
         $form->handleRequest($request);
 
+        if ($form->isSubmitted()) {
+            if ($form->isValid()) {
+                $researchGroupId = $request->get('ResearchGroupSelector');
+                return $this->getReport($researchGroupId);
+            }
+        }
+
+        if ($id) {
+            return $this->getReport($id);
+        }
```

Comment:
Can we make this into a big statement?
```suggestion
        if (($form->isSubmitted() and if ($form->isValid()) or $researchGroupId ) {
            ) {
                $researchGroupId = $request->get('ResearchGroupSelector');
                return $this->getReport($researchGroupId);
            }
        }
```

Reply:
You also have to change id in the route above, to researchGroupId. FYI, don't commit this, it may not work.

Reply:
I think it is good as it is, or else we need to change the variable name in the route config and add an additional check for it